Mathlib/Probability/Kernel/Composition/CompProd.lean

```diff
@@ -56,141 +56,67 @@ We define a kernel composition-product
-    ext1 b
-    rw [measure_iUnion]
-    · intro i j hij s hsi hsj c hcs
-      have hbci : {(b, c)} ⊆ f i := by rw [Set.singleton_subset_iff]; exact hsi hcs
-      have hbcj : {(b, c)} ⊆ f j := by rw [Set.singleton_subset_iff]; exact hsj hcs
-      simpa only [Set.bot_eq_empty, Set.le_eq_subset, Set.singleton_subset_iff,
-        Set.mem_empty_iff_false] using hf_disj hij hbci hbcj
-    · exact fun i ↦ measurable_prodMk_left (hf_meas i)
-  rw [h_tsum, lintegral_tsum]
-  · simp [compProdFun]
-  · intro i
-    have hm : MeasurableSet {p : (α × β) × γ | (p.1.2, p.2) ∈ f i} :=
-      (hf_meas i).preimage (by fun_prop)
-    exact ((measurable_kernel_prodMk_left hm).comp measurable_prodMk_left).aemeasurable
-
-theorem compProdFun_tsum_right (κ : Kernel α β) (η : Kernel (α × β) γ) [IsSFiniteKernel η] (a : α)
-    (hs : MeasurableSet s) : compProdFun κ η a s = ∑' n, compProdFun κ (seq η n) a s := by
-  simp_rw [compProdFun, (measure_sum_seq η _).symm]
-  have : ∫⁻ b, Measure.sum (fun n => seq η n (a, b)) {c : γ | (b, c) ∈ s} ∂κ a
-      = ∫⁻ b, ∑' n, seq η n (a, b) {c : γ | (b, c) ∈ s} ∂κ a := by
-    congr with b
-    rw [Measure.sum_apply]
-    exact measurable_prodMk_left hs
-  rw [this, lintegral_tsum]
-  exact fun n ↦ ((measurable_kernel_prodMk_left (κ := (seq η n))
-    ((measurable_fst.snd.prodMk measurable_snd) hs)).comp measurable_prodMk_left).aemeasurable
-
-theorem compProdFun_tsum_left (κ : Kernel α β) (η : Kernel (α × β) γ) [IsSFiniteKernel κ] (a : α)
-    (s : Set (β × γ)) : compProdFun κ η a s = ∑' n, compProdFun (seq κ n) η a s := by
-  simp_rw [compProdFun, (measure_sum_seq κ _).symm, lintegral_sum_measure]
-
-theorem compProdFun_eq_tsum (κ : Kernel α β) [IsSFiniteKernel κ] (η : Kernel (α × β) γ)
-    [IsSFiniteKernel η] (a : α) (hs : MeasurableSet s) :
-    compProdFun κ η a s = ∑' (n) (m), compProdFun (seq κ n) (seq η m) a s := by
-  simp_rw [compProdFun_tsum_left κ η a s, compProdFun_tsum_right _ η a hs]
-
-theorem measurable_compProdFun (κ : Kernel α β) [IsSFiniteKernel κ] (η : Kernel (α × β) γ)
-    [IsSFiniteKernel η] (hs : MeasurableSet s) :
-    Measurable fun a ↦ compProdFun κ η a s := by
-  simp only [compProdFun]
-  have h_meas : Measurable (Function.uncurry fun a b => η (a, b) {c : γ | (b, c) ∈ s}) := by
-    have : (Function.uncurry fun a b => η (a, b) {c : γ | (b, c) ∈ s})
-        = fun p ↦ η p {c : γ | (p.2, c) ∈ s} := by
-      ext1 p
-      rw [Function.uncurry_apply_pair]
-    rw [this]
-    exact measurable_kernel_prodMk_left (measurable_fst.snd.prodMk measurable_snd hs)
-  exact h_meas.lintegral_kernel_prod_right
-
-open scoped Classical in
 /-- Composition-Product of kernels. For s-finite kernels, it satisfies
 `∫⁻ bc, f bc ∂(compProd κ η a) = ∫⁻ b, ∫⁻ c, f (b, c) ∂(η (a, b)) ∂(κ a)`
 (see `ProbabilityTheory.Kernel.lintegral_compProd`).
 If either of the kernels is not s-finite, `compProd` is given the junk value 0. -/
 noncomputable def compProd (κ : Kernel α β) (η : Kernel (α × β) γ) : Kernel α (β × γ) :=
```

Comment:
Maybe make it an `irreducible_def`? It's complicated enough that we don't want the kernel to get lost on it.